src/lib/ngsi/ContextAttribute.cpp

```diff
@@ -371,10 +371,17 @@ std::string ContextAttribute::renderAsJsonObject
         break;
 
       case ValueTypeNumber:
-        char num[32];
-        snprintf(num, sizeof(num), "%f", numberValue);
-        effectiveValue      = std::string(num);
-        valueIsNumberOrBool = true;
+        if (type == DATE_TYPE)
+        {
+          effectiveValue = isodate2str(numberValue);
+        }
+        else // regular number
+        {
+          char num[32];
```

Comment:
32? :-1 :-)

There should a constant for this is limits.h:

#define STRING_SIZE_FOR_DOUBLE  32    // Room enough for a double

Or something like  that.
32 should be enough, right?